src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapper.java

```diff
@@ -671,18 +675,37 @@ private void buildConsentExtensionFilingDefence(EventHistory.EventHistoryBuilder
         if (dateReceived == null) {
             return;
         }
+        // date and time check to find the login
+        LocalDate extensionDate = caseData.getRespondentSolicitor1AgreedDeadlineExtension();
+
+        //finding extension date for the correct respondent in a 1v2 different solicitor scenario
+        MultiPartyScenario multiPartyScenario = getMultiPartyScenario(caseData);
+        if (multiPartyScenario == ONE_V_TWO_TWO_LEGAL_REP) {
+            if ((caseData.getRespondent1TimeExtensionDate() == null)
+                && (caseData.getRespondent2TimeExtensionDate() != null)) {
+                extensionDate = caseData.getRespondentSolicitor2AgreedDeadlineExtension();
+            } else if ((caseData.getRespondent1TimeExtensionDate() != null)
+                && (caseData.getRespondent2TimeExtensionDate() != null)) {
+                if (caseData.getRespondent2TimeExtensionDate()
+                    .isAfter(caseData.getRespondent1TimeExtensionDate())) {
+                    extensionDate = caseData.getRespondentSolicitor2AgreedDeadlineExtension();
+                } else {
+                    extensionDate = caseData.getRespondentSolicitor1AgreedDeadlineExtension();
+                }
+            }
+        }
+
```

Comment:
Test cases will need to be added for this file.
In relation to the `dateReceived` field in this method, it is currently specific to respondent 1 only.. What if respondent2 is asking for a time extension, is this date being stored somewhere?